File: python/algoritmos_basicos/mcm.py
#  Mínimo Comum Múltiplo (MCM) Recursivo

def mcmR(n, m, i=1):
    if (n * i) % m == 0:
        return n * i
    return mcmR(n, m, i + 1)

print(mcmR(2, 12))
#  Mínimo Comum Múltiplo (MCM) Iterativo
def mcmI(n,m, i = 1):
    while True:
        if (n * i) % m == 0:
            return n * i
        i += 1

print(mcmI(32, 13))

File: c++/Algoritmos Básicos/mcm.cpp
#include <iostream>

using namespace std;

int mcmR(int n, int m, int i = 1)
{
    if ((n * i) % m == 0)
    {
        return n * i;
    }
    return mcmR(n, m, i + 1);
}

int mcmI(int n, int m, int i = 1)
{
    while (true)
    {
        if ((n * i) % m == 0)
        {
            return n * i;
        }
        i += 1;
    }
}
int main()
{
    cout << mcmR(2, 12) << endl;
    cout << "\n"
         << endl;
    cout << mcmI(32, 13) << endl;

    return 0;
}